.github/workflows/pages_deploy_main.yaml

```diff
@@ -0,0 +1,58 @@
+name: Documentation Deploy
+
+on:
+  # When a PR is merged (or force push to main)
+  push:
+    branches: ["main"]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Enable cargo logs to be in colour
+env:
+  CARGO_TERM_COLOR: always
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: write
```

Comment:
TODO: make read @george-cosma

Reply:
https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#overview

> For each of the available permissions, shown in the table below, you can assign one of the access levels: read, write, or none. _**`write includes read.`**_ If you specify the access for any of these permissions, all of those that are not specified are set to none